### PR TITLE
Preliminary support for Ubuntu 24 and Ubuntu 24 ARM

### PR DESCRIPTION
**Pull request by itsarune**
### Description

Summary of changes:
- python stepping to python 3.12 (required to install pyqt6 on ARM devices via the system package, pip install was broken)
- updated SSL game controller to latest (for ARM binaries)
- updated Tigers Autoref to latest (new game controller also updates several protobuf definitions)

- What works:
- ARM bazel build works
- X86 full_system/thunderscope/jetson compilation works

~~What doesn't work:~~
- jetson nano compilation for ARM (should work)

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: docs/getting-started-wsl.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ If you are not using Windows 11 and would prefer not to upgrade, you can follow
 3. Now, let's install Ubuntu.
     - Download the WSL2 kernel from [here](https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel).
     - Open a PowerShell window and run command `wsl --set-default-version 2` to use WSL2 by default.
-    - Install Ubuntu 20.04 LTS or Ubuntu 22.04 LTS from the Microsoft Store.
+    - Install Ubuntu 20.04 LTS, Ubuntu 22.04 LTS and Ubuntu 24.04 LTS from the Microsoft Store.
```
Comment (williamckha) [inline docs/getting-started-wsl.md]: typo? should be or

File: docs/getting-started.md [hunk 2]
```diff
@@ -63,9 +63,15 @@ These instructions assume you have a basic understanding of Linux and the comman
 
 ### Operating systems
 
-We currently only support Linux, specifically Ubuntu 20.04 LTS and Ubuntu 22.04 LTS. You are welcome to use a different version or distribution of Linux, but may need to make some tweaks in order for things to work.
+We currently only support Linux, specifically Ubuntu.
 
-You can use Ubuntu 20.04 LTS and Ubuntu 22.04 LTS inside Windows through Windows Subsystem for Linux, by following [this guide](./getting-started-wsl.md). **Running and developing Thunderbots on Windows is experimental and not officially supported.**
+If you have a X86_64 machine, we support Ubuntu 20.04 LTS, Ubuntu 22.04 LTS and Ubuntu 24.04 LTS.
+
+If you have a AARCH64 machine, we support Ubuntu 24.04 LTS.
```
Comment (williamckha) [inline docs/getting-started.md]: should add in brackets (ARM64) because AArch64 is not as recognizable

**Comment by williamckha** [comment]
Maybe while you're at it, throw in #2592

File: docs/getting-started-wsl.md (hunk 1 above)
Comment (itsarune) [reply docs/getting-started-wsl.md]: updated

File: docs/getting-started.md (hunk 2 above)
Comment (itsarune) [reply docs/getting-started.md]: updated

**Comment by itsarune** [comment]
I'm too lazy to double-check power board and estop with the nanopb upgrade, so I'm just going to leave it off.

**Comment by itsarune** [comment]
Could someone with an ARM computer run the following command and let me know if it works: `bazel build //software/embedded:thunderloop_main --copt=-O3 --//software/embedded:host_platform=NANO --platforms=//cc_toolchain:robot`

**Comment by annieisawesome2** [comment]
I gave it a try - it didn't work:

ERROR: Skipping '//software/embedded:host_platform': no such package 'software/embedded': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.

**Comment by Lmh-java** [comment]
It worked on my machine. 
```
global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.'
   36 | BOOST_PRAGMA_MESSAGE(
      | ^~~~~~~~~~~~~~~~~~~~
Target //software/embedded:thunderloop_main up-to-date:
  bazel-bin/software/embedded/thunderloop_main
INFO: Elapsed time: 236.995s, Critical Path: 83.25s
INFO: 189 processes: 49 disk cache hit, 21 internal, 119 linux-sandbox.
INFO: Build completed successfully, 189 total actions
```

**Review by williamckha — APPROVED** [review]
🚀

**Comment by Mr-Anyone** [comment]
I have tried a few entries in thunderscope and it worked!

One thing to note when running autoref is that a warning message is spammed: 

```
sys:1: UserWarning: MessageFactory class is deprecated. Please use GetMessageClass() instead of MessageFactory.GetPrototype. MessageFactory class will be removed after 2024.
sys:1: UserWarning: MessageFactory class is deprecated. Please use GetMessageClass() instead of MessageFactory.GetPrototype. MessageFactory class will be removed after 2024.
sys:1: UserWarning: MessageFactory class is deprecated. Please use GetMessageClass() instead of MessageFactory.GetPrototype. MessageFactory class will be removed after 2024.
sys:1: UserWarning: MessageFactory class is deprecated. Please use GetMessageClass() instead of MessageFactory.GetPrototype. MessageFactory class will be removed after 2024.
```

This seems to be a google protobuf python warning


https://github.com/protocolbuffers/protobuf/blob/b8d356701a788951d92a1b4f04881fb539ca62d7/python/google/protobuf/message_factory.py#L136-L137

**Comment by Mr-Anyone** [comment]
Apart from that, LGTM!

**Review by Mr-Anyone — APPROVED** [review]

**Comment by itsarune** [comment]
Yeah, I saw this error too. We aren't using `MessageFactory` but some dependency of ours is. If you're okay with it, we'll handle that in a different ticket?

**Comment by Mr-Anyone** [comment]
perfectly fine with it

**Comment by Mr-Anyone** [comment]
deal with it once it will be a issues

**Pull request merged**